resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/ProviderHelper.java

```diff
@@ -155,24 +155,15 @@ public static void writeTo(final InputStream in, final OutputStream out) throws
     * @param out output stream
     * @throws IOException if I/O error occurred
     */
-   public static CompletionStage<Void> writeTo(final InputStream in, final AsyncOutputStream out)
-   {
-      final byte[] buf = new byte[2048];
-      return writeTo(in, out, buf);
-   }
-
-   private static CompletionStage<Void> writeTo(final InputStream in, final AsyncOutputStream out, byte[] buf)
-   {
+   public static CompletionStage<Void> writeTo(final InputStream in, final AsyncOutputStream out) {
+      byte[] buf = new byte[2048];
       int read;
-      try
-      {
-         if ((read = in.read(buf)) != -1)
-         {
-            return out.asyncWrite(buf, 0, read)
-                  .thenCompose(v -> writeTo(in, out, buf));
+      try {
+         while ((read = in.read(buf)) != -1) {
+            out.asyncWrite(buf, 0, read);
+            buf = new byte[2048];

```

Comment:
Maybe I'm wrong here, but isn't this keeping on allocating new byte arrays? Can't a single one be re-used?

Reply:
+1 we shouldn't need to reinitialize the array.